client/src/document/hooks.ts

```diff
@@ -8,3 +11,79 @@ export function useDocumentURL() {
   // pathname with the document's mdn_url.
   return url.endsWith("/") ? url.substring(0, url.length - 1) : url;
 }
+
+export function useCopyExamplesToClipboard(doc: Doc | undefined) {
+  React.useEffect(() => {
+    if (!doc) {
+      return;
+    }
```

Comment:
Just like you have that `copyToClipboard` function available, I'd love it if we can write one called something like `isClipboardSupported`. And then add:

```suggestion
    }
    
    if (!isClipboardSupported()) {
      console.log("Copy-to-clipboard disabled because your browser does not appear to support it");
      return;
    }
```

Reply:
I'd like this too. Nice one!